src/views/instances/new-dhis2/fields/name-input.tsx

```diff
@@ -9,7 +9,7 @@ import styles from './fields.module.css'
  * restrictions on the length. Labels must be 63 characters or less. */
 const startPattern = /^[a-z]/
 const endPattern = /[a-z0-9]$/
-const interiorPattern = /[-a-z0-9]*/
+const interiorPattern = /[^-a-z0-9]+/
```

Comment:
Just a note - I thought that negating the pattern here and removing the negation from the `if` statement below is logically the same, but since we're not using an exact match (with a pattern start and end - `^.+$`), a name like `test-$$$-name` will be matched by `/[-a-z0-9]*/` (`test-` and `-name` match the pattern) , which we don't want.
![image](https://github.com/dhis2-sre/im-web-client/assets/8098421/762e9563-28a4-49f9-89b0-8d8e9ce7fb92)


On the other hand, `$$$` in `test-$$$-name` will be matched by `/[^-a-z0-9]+/` and return an error as expected.
![image](https://github.com/dhis2-sre/im-web-client/assets/8098421/daff757a-d31b-4ae7-8126-c3fd826e1e45)